### PR TITLE
Fix/issue 5693 firefox focus trap

### LINKED ISSUE #5693: Button block acts as keyboard trap on Firefox
## Steps to Reproduce (for bugs)

1. Open the demo post on Firefox (I am using Firefox developer edition, which is currently numbered at Firefox 60).
2. Insert cursor in a paragraph block, and navigate down the page with the down arrow key.
3. Once you reach the button block the URL is focussed, with the carat before the http:// and the keyboard down arrow does not work. You have to tab into the next block, which is jaring for anyone expecting to be able to use their arrow keys.

I am using Windows.

## Expected Behavior

All editors that I can remember using recently have allowed moving between all text blocks using the left/ right/ up/ down arrow keys alone. This behaivour should be fully supported in Gutenburg.

## Todos
- [ ] Tests
- [ ] Documentation

### PR DESCRIPTION
**Pull request by timwright12**
## Description
Fixes https://github.com/WordPress/gutenberg/issues/5693

## How has this been tested?
Browser this fix in tested in Chrome, Safari, Firefox, Windows Firefox

## Types of changes
Interrupted the key event on inputURL if the caret position isn't at the beginning or end of the input field text to mimic the expected behavior in non-Win FF browsers

## Checklist:
- [ x ] My code is tested.
- [ x ] My code follows the WordPress code style. 
- [ x ] My code follows the accessibility standards. 
- [ x ] My code has proper inline documentation.

### CODE REVIEW

**Comment by timwright12** [comment]
@tofumatt Please let me know if anything is "off" with the PR as it's pretty much my first one, happy to update or run anything locally that I forgot to to get tests passing or whatever!

**Review by tofumatt — CHANGES_REQUESTED** [review]
Thanks for the PR @timwright12! A few meta points around the PR:

* no need to reference this PR in the original issue; GitHub does that for us so it's just extra notifications/noise that isn't needed 😄
* You can use "Fixes #xyz" in the issue description so the issue you're working on fixing will be closed automatically once the patch lands
* You should remove the addition of the `gutenberg-mobile` submodule; that was removed awhile back and shouldn't be committed 😄

Otherwise the process around everything looks good 👍

This looks good but I think could use comments that clarify things a bit more; right now this code would be a bit confusing to a developer unaware of this bug or the reason for this code to exist.

Normally I'd also advocate for an E2E (end-to-end) test here, but our E2E tests run in Chromium on Linux so 🤷‍♂️ Since the code path is still the same it might be nice to test it and make sure that keyboard presses work as intended, but we won't really be testing the browser/OS combo that caused the bug in the first place…

Thanks for the patch, let me know if anything needs clarification, but I think after some comment tweaks this'll be good to merge! 👍

File: packages/editor/src/components/url-input/index.js
Comment (timwright12) [reply packages/editor/src/components/url-input/index.js]: @tofumatt honestly, for this one, I was just following the pattern from the rest of the file. preventDefault() is probably enough. Do you want me to update that or leave it to match the previous code?

Comment (tofumatt) [reply packages/editor/src/components/url-input/index.js]: Oh, fair enough; if it's everywhere else in the file we can do that and then look at that later...

Comment (timwright12) [reply packages/editor/src/components/url-input/index.js]: Ok cool, I updated the commenting and removed the submodule, please let me know if there's anything else!

**Review by tofumatt — APPROVED** [review]
Awesome work, thanks for the clarification in the comment. Really sorry about the delay in getting to this review.

I've tweaked the comments just a little bit, all good to ship. Thanks again! 👍

**Pull request merged**
